pkg/artifact/sbom/cyclonedx.go

```diff
@@ -0,0 +1,109 @@
+package sbom
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	digest "github.com/opencontainers/go-digest"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/analyzer/config"
+	"github.com/aquasecurity/fanal/artifact"
+	"github.com/aquasecurity/fanal/cache"
+	"github.com/aquasecurity/fanal/handler"
+	"github.com/aquasecurity/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/sbom/cyclonedx"
+)
+
+var (
+	ArtifactCycloneDX types.ArtifactType = "cyclonedx"
+)
+
+type Artifact struct {
+	filePath       string
+	cache          cache.ArtifactCache
+	analyzer       analyzer.AnalyzerGroup
+	handlerManager handler.Manager
+
+	artifactOption      artifact.Option
+	configScannerOption config.ScannerOption
+}
```

Comment:
We can define an interface like

```
type Parser interface {
    Parse(io.Reader) *ftypes.OS, []ftypes.PackageInfo, []ftypes.Application, error
}
```

Then, we can have a common artifact over all SBOM specs.

```suggestion
type Artifact struct {
	filePath       string
	cache          cache.ArtifactCache
	sbomParser       Parser
	analyzer       analyzer.AnalyzerGroup
	handlerManager handler.Manager

	artifactOption      artifact.Option
	configScannerOption config.ScannerOption
}
```

And rename cyclonedx.go to sbom.go.

Reply:
Add parser.go 

https://github.com/aquasecurity/trivy/pull/2203/files#diff-c412642f14eedf5814cd3a2052e78fa7bfdb109b7e1f0af98b6e356bd21eb498